### PR TITLE
fix(curriculum): test both stories in Sentence Maker

### PR DESCRIPTION
**Pull request by gikf**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

- - -
- Tests both stories in separate tests.

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/lab-sentence-maker/66c057041df6394ca796bf33.md [hunk 1]
```diff
@@ -119,11 +119,11 @@ You should declare a `firstStory` variable.
 assert.isNotNull(firstStory);
 ```
 
-Your should use the correct story format for both stories: `"Once upon a time, there was a(n) [adjective] [noun] who loved to eat [noun2]. The [noun] lived in a [place] and had [adjective2] nostrils that blew fire when it was [verb]."`. Pay attention to spaces.
+Your should use the correct story format for first story: `"Once upon a time, there was a(n) [adjective] [noun] who loved to eat [noun2]. The [noun] lived in a [place] and had [adjective2] nostrils that blew fire when it was [verb]."`. Pay attention to spaces.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-sentence-maker/66c057041df6394ca796bf33.md:122]:
```suggestion
You should use the correct story format for the first story: `"Once upon a time, there was a(n) [adjective] [noun] who loved to eat [noun2]. The [noun] lived in a [place] and had [adjective2] nostrils that blew fire when it was [verb]."`. Pay attention to spaces.
```

File: curriculum/challenges/english/25-front-end-development/lab-sentence-maker/66c057041df6394ca796bf33.md [hunk 2]
```diff
@@ -176,6 +176,13 @@ You should reassign the `noun2` variable for the second story.
 assert.lengthOf(__helpers.removeJSComments(code).match(/noun2\s*=\s*/g), 2);
 ```
 
+Your should use the correct story format for second story: `"Once upon a time, there was a(n) [adjective] [noun] who loved to eat [noun2]. The [noun] lived in a [place] and had [adjective2] nostrils that blew fire when it was [verb]."`. Pay attention to spaces.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-sentence-maker/66c057041df6394ca796bf33.md:179]:
```suggestion
You should use the correct story format for the second story: `"Once upon a time, there was a(n) [adjective] [noun] who loved to eat [noun2]. The [noun] lived in a [place] and had [adjective2] nostrils that blew fire when it was [verb]."`. Pay attention to spaces.
```

**Review by Dario-DC — CHANGES_REQUESTED** [review]
I started to review but then I realized that this doesn't work since the variables `adjective`, `noun`, `verb`, `place`, `adjective2`, and `noun2` are reassigned. So you cannot check the first story without regex. This is the reason for which only the second story is tested.

The main issue is that this lab is in the JS functions section. Therefore, should use functions. This was discussed and it never done, apparently. Just opened an issue #57710

**Comment by gikf** [comment]
> I started to review but then I realized that this doesn't work since the variables `adjective`, `noun`, `verb`, `place`, `adjective2`, and `noun2` are reassigned. So you cannot check the first story without regex. This is the reason for which only the second story is tested.

At the point of re-assigning, the `firstSentence` is already defined, so it will have string with values filled from variables.

**Comment by Dario-DC** [comment]
One test fails because the expected string contains the reassigned values

```
Test text: <p>Your should use the correct story format for first story: <code>"Once upon a time, there was a(n) [adjective] [noun] who loved to eat [noun2]. The [noun] lived in a [place] and had [adjective2] nostrils that blew fire when it was [verb]."</code>. Pay attention to spaces.</p>
      + expected - actual

      -Once upon a time, there was a(n) funny dragon who loved to eat cakes. The dragon lived in a garden and had sparkling nostrils that blew fire when it was jumping.
      +Once upon a time, there was a(n) cute puppy who loved to eat flower. The puppy lived in a park and had colorful nostrils that blew fire when it was barking.
      
  AssertionError: expected 'Once upon a time, there was a(n) funn…' to equal 'Once upon a time, there was a(n) cute…'
```

**Comment by ilenia-magoni** [comment]
> At the point of re-assigning, the `firstSentence` is already defined, so it will have string with values filled from variables.

but `expected` is built with the current values of the variables, not with the previous ones

**Comment by gikf** [comment]
Ah! 🤦‍♂️ I wasn't thinking about what happens in the test. I think we could capture values initially assigned with regex, to still test the _complete_ sentence.

**Comment by Dario-DC** [comment]
> Ah! 🤦‍♂️ I wasn't thinking about what happens in the test. I think we could capture values initially assigned with regex, to still test the _complete_ sentence.

I don't see other options :+1:

**Review by Dario-DC — APPROVED** [review]
Awesome :tada:

**Review by a2937 — APPROVED** [review]
While there is an (albeit low) chance of someone doing something silly changing only one value and reassigning the rest to the old ones, it's not a major concern. And while yes they can assign values to the variables multiple times; they should be able to tell from the hints that the variable is meant to be assigned to only twice.

**Pull request merged**